recipe-server/client/control/components/recipes/RecipeForm.js

```diff
@@ -57,6 +57,10 @@ export default class RecipeForm extends React.PureComponent {
     'opt-out-study': OptOutStudyFields,
   };
 
+  componentDidMount() {
+    this.defaultIdenticonSeed = this.defaultIdenticonSeed || IdenticonField.generateSeed();
```

Comment:
Should `defaultIdenticonSeed` be in state? It seems that since it is used in render(), if it changes it should cause a re-render. If not, it should be set to some value (such as null) in a constructor, which helps a) performance and b) readability.

Reply:
I actually specifically _didn't_ want a re-render. Essentially, a bug I ran into was that on each render, the random 'default' seed would change, causing a weird identicon swap on page load (showed one first, then the other). Setting the `defaultIdenticonSeed` on the actual instance was meant to avoid the additional re-render (vs using state).

I've moved this `defaultIdenticonSeed` declaration to a constructor instead!

Reply:
I guess I'm still not getting something. Why do we store a default at all? Couldn't we leave the `initialValue` of the form item empty, and let it sort itself out? This seems like complexity that `IdenticonField` should be handling.

Also, if we keep it this way, the `this.defaultIdenticonSeed` will always be undefined in the constructor, so the set-if-not-exists dance isn't useful there.

Reply:
The identicon_seed field has its value set up like so:`value={recipe.get('identicon_seed')}` - the issue here is that for new/cloned recipes, the `identicon_seed` will be undefined, but we need a random one to be chosen so it appears on the 'create' form.

Doing something like `recipe.get('identicon', generateRandomIcon())` solves this in that it falls back to a random identicon if one is not set. But, the issue there is that on each re-render, `generateRandomIcon` is fired, so the identicon changes with each render. To solve that, I used `this.defaultSeed` - it's set once, and then remains that value for the rest of the component's lifespan. Subsequent re-renders still fall back on the already defined default.

The issue with having `IdenticonField` generating a default is the case in which the field does not have a value but will have one loaded later. Since it's a simple form component, value changes need to be registered through `onChange` (that is, the `value` is tracked as a prop, not internally in the component). From there, it steps into a weird territory where it fires `onChange` when there's no value yet, but if another value is loaded, then another `onChange` needs to fire. This leads to weird flashes of different identicons appearing during page load. And, speaking semantically, since it really wasn't changing but being set to an initial value, an `onChange` event shouldn't technically need to fire.

The reason why I had originally put the `defaultIdenticonSeed` initialization inside `didMount` was the idea that on each mount, the default seed identicon would change, preventing the same default identicon from being offered (say, if the user was creating multiple recipes in one session). I reckon the code being in the constructor function the same, with the only difference being we need to add the `super(props)` boilerplate to the constructor. You are correct about not needing the `thing = thing || otherThing` business in the constructor, though.

Reply:
Thanks for that explanation. I think I have a good handle of what's going on now. Who knew that randomized fields would be so complicated.

After all that, I think that setting the default in `didMount`, without preserving existing values (`this.defaultIdenticon = generateRandomIcon()`) would probably work well. It wouldn't affect recipes with existing identicons, and it would work even if component instances got re-used. (Is that a thing we need to worry about?)

That being said, I'm ok with any of the options we've gone through in this PR. If it fixes the clone bug, and doesn't flicker, I don't have any other concerns, and it seems like all of your solutions achieve that. I appreciate you working through it with me.